### PR TITLE
`ExternalMessageEnvelope`: documentation and improved coverage

### PR DESCRIPTION
**Pull request by armiol**
This PR addresses #638 and #639.

In particular it clarifies the API of `ExternalMessageEnvelope`. 

Also it improves the test coverage of `DelegatingEventDispatcher` behavior, that dispatches `ExternalMessageEnvelope` instances.

As a bonus, `RejectionHandlerMethod` documentation was improved with explicit mentioning of `@React` and `@Subscribe` in different cases.

### CODE REVIEW

**Comment by armiol** [comment]
@alexander-yevsyukov PTAL.

File: server/src/main/java/io/spine/server/rejection/RejectionHandlerMethod.java [hunk 1]
```diff
@@ -152,11 +152,18 @@ Object doInvoke(Object target, Message rejectionMsg, RejectionContext context) {
          * A rejection handler method which receives a rejection message
          * as a single parameter.
          *
-         * <p>The signature of such a method is following:
+         * <p>The signature of such a method is as follows, if used in {@link RejectionSubscriber}:
          * <pre>
-         * [{@literal @}HandlerAnnotation]
+         * [{@literal @}Subscribe]
```
Comment (alexander-yevsyukov) [inline server/src/main/java/io/spine/server/rejection/RejectionHandlerMethod.java]: Why do we need square brackets around `@Subscribe` here and in the other Javadocs below?

Comment (armiol) [reply server/src/main/java/io/spine/server/rejection/RejectionHandlerMethod.java]: @alexander-yevsyukov they were previously used to mark `[HandlerAnnotation]`.

Removed.

**Comment by armiol** [comment]
@alexander-yevsyukov PTAL.

**Pull request merged**